refresh.template.py

```diff
@@ -90,6 +97,57 @@ def _get_headers(compile_args: typing.List[str], source_path_for_sanity_check: t
     return headers
 
 
+def _get_headers_msvc(compile_args: typing.List[str]):
+    # We don't want to compile, but only pre-process and show all the included header files.
+    # We don't want to have the preprocessed results written to a file, so we get them printed
+    # to `stderr`.
+    header_cmd = list(compile_args) + ["/showIncludes", "/EP"]
+
+    # cl.exe expects `INCLUDE` (and normally `LIB` but that is irrelevant here) to be set so that it can find the
+    # system headers. We have no idea what the correct paths are. So in case these variables are not set, we need
+    #  to infer them somehow. These variables might not be set if the user has a custom `cc_toolchain()`
+    # configured with `bazel` and does not rely on having sourced vcvarsall.bat.
+    environ = dict(os.environ)
+    if "INCLUDE" not in environ:
+        # We assume the base MSVC install based based on the `cl.exe` path in our command.
+        base = pathlib.Path(header_cmd[0]).parent.parent.parent.parent
+        # And we find the newest Windows 10 kits version that exists.
+        windows_kits = sorted( pathlib.Path("C:\\Program Files (x86)\\Windows Kits\\10\\include").glob("*"))[-1]
+        environ["INCLUDE"] =  os.pathsep.join(
+            (
+                os.fspath(base / "ATLMFC/include"),
+                os.fspath(base / "include"),
+                os.fspath(windows_kits / "ucrt"),
+                os.fspath(windows_kits / "shared"),
+                os.fspath(windows_kits / "um"),
+                os.fspath(windows_kits / "winrt"),
+                os.fspath(windows_kits / "cppwinrt"),
+            )
+        )
```

Comment:
Phew. I remember your saying that it needed this. Thanks for finding a way through.
[I'm assuming this means that Bazel didn't pass the environment variable in aquery? I think that's worth filing a Bazel issue about.]

Questions that occur to me, not knowing that much about Windows: 
- Will `base` work with clang-cl?
- Will the `10` path work on other versions of Windows?

Reply:
I don't know the answers to these questions. I am a linux guy. I just have to support some developers on windows.

Reply:
I'd really like it if we could make this more robust somehow